### PR TITLE
Update Kotlin and Compose to v2

### PR DESCRIPTION
**Pull request by jrodbx**
Closes #1441

### CODE REVIEW

File: paparazzi/build.gradle [hunk 1]
```diff
@@ -45,7 +45,7 @@ dependencies {
   api libs.junit
   api libs.androidx.annotations
   api libs.guava
-  api libs.kotlinx.coroutines.core
+  api libs.kotlinx.coroutines.android
```
Comment (jrodbx) [inline paparazzi/build.gradle:48]: it appears that there used to be a transitive dependency on coroutines.android and this bump caused this reference to [Handler.asCoroutineDispatcher](https://github.com/cashapp/paparazzi/blob/master/paparazzi/src/main/java/app/cash/paparazzi/PaparazziSdk.kt#L588) to fail.

File: gradle/libs.versions.toml [hunk 2]
```diff
@@ -73,8 +74,9 @@ plugin-android = { module = "com.android.tools.build:gradle", version.ref = "agp
 plugin-buildConfig = { module = "com.github.gmazzo.buildconfig:plugin", version = "5.3.5" }
 plugin-dokka = { module = "org.jetbrains.dokka:dokka-gradle-plugin", version = "1.9.20" }
 plugin-kotlin = { module = "org.jetbrains.kotlin:kotlin-gradle-plugin", version.ref = "kotlin" }
+plugin-kotlin-compose = { module = "org.jetbrains.kotlin:compose-compiler-gradle-plugin", version.ref = "kotlin" }
 plugin-kotlinApiDump = { module = "org.jetbrains.kotlinx:binary-compatibility-validator", version = '0.14.0' }
 plugin-ksp = { module = "com.google.devtools.ksp:com.google.devtools.ksp.gradle.plugin", version.ref = "ksp" }
 plugin-mavenPublish = { module = "com.vanniktech:gradle-maven-publish-plugin", version = "0.28.0" }
-plugin-poko = { module = "dev.drewhamilton.poko:poko-gradle-plugin", version = "0.15.3" }
+plugin-poko = { module = "dev.drewhamilton.poko:poko-gradle-plugin", version = "0.16.0-beta01" }
```
[the remark this answers is not in the document]
Comment (JakeWharton) [reply gradle/libs.versions.toml:81]: https://github.com/drewhamilton/Poko/issues/352

File: gradle/libs.versions.toml [hunk 3]
```diff
@@ -27,14 +27,15 @@ androidx-recyclerview = { module = "androidx.recyclerview:recyclerview", version
 bytebuddy-agent = { module = "net.bytebuddy:byte-buddy-agent", version.ref = "bytebuddy" }
 bytebuddy-core = { module = "net.bytebuddy:byte-buddy", version.ref = "bytebuddy" }
 
-compose-compiler = { module = "androidx.compose.compiler:compiler", version.ref = "composeCompiler" }
+compose-compiler = { module = "org.jetbrains.kotlin:kotlin-compose-compiler-plugin-embeddable", version.ref = "kotlin" }
```
Comment (JakeWharton) [inline gradle/libs.versions.toml:30]: This should not be needed anymore. Apply the Compose plugin instead.

Comment (jrodbx) [reply gradle/libs.versions.toml:30]: I tried that, but paparazzi is a jvm module, so this line was failing when applying the Compose plugin:

```
add(AbstractKotlinPluginKt.PLUGIN_CLASSPATH_CONFIGURATION_NAME, libs.compose.compiler)
```

**Review by JakeWharton — APPROVED** [review]

File: gradle/libs.versions.toml (hunk 3 above)
Comment (JakeWharton) [reply gradle/libs.versions.toml:30]: Delete that line completely. The plugin will work fine in JVM modules.

Comment (jrodbx) [reply gradle/libs.versions.toml]: Ah, you're right.  I guess my issue here was related to the plugin test fixture configuration setup that I later fixed.  Good catch!

**Pull request merged**